kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/ssl/SslManagerServiceImpl.java

```diff
@@ -629,27 +629,20 @@ private boolean changeKeyStorePassword() {
             }
         }
 
-        if (oldPassword == null) {
-            logger.warn("null old password");
-            return false;
-        }
-
-        if (!Arrays.equals(oldPassword, newPassword)) {
+        if (oldPassword != null && !Arrays.equals(oldPassword, newPassword)) {
             try {
                 if (isKeyStoreAccessible(this.options.getSslKeyStore(), oldPassword)) {
                     changeKeyStorePassword(this.options.getSslKeyStore(), oldPassword, newPassword);
                 } else if (isKeyStoreAccessible(this.options.getSslKeyStore(), newPassword)) {
                     changeKeyStorePassword(this.options.getSslKeyStore(), newPassword, newPassword);
```

Comment:
What is the purpose of changing the keystore password with itself?

Reply:
If I'm not wrong was related to a keystore just provided through the web ui where you don't have an old password and you need the new password both to access the keystore and register it in the snapshot.

Reply:
OK, but I don't get what `changeKeyStorePassword(this.options.getSslKeyStore(), newPassword, newPassword);` does

Reply:
checks that the keystore is accessible and stores the new password

Reply:
I mean this:
```
else if (isKeyStoreAccessible(this.options.getSslKeyStore(), newPassword)) {
                     changeKeyStorePassword(this.options.getSslKeyStore(), newPassword, newPassword);
```
It looks weird to me as it sounds "if the keystore is already accessible with newPassword, change the password (which is newPassword) to to newPassword"

Reply:
it is definitely weird. But, if you remember, we need to structure the code in this way the last time we reviewed it together to deal with the different requirements we have. For sure it needs to be improved, but it was not the scope of this PR also because we are close to the release.